lib/sisimai/bite/email/mailfoundry.rb

```diff
@@ -0,0 +1,133 @@
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['subject'] =~ Re0[:subject]
+        return nil unless mhead['received'].find { |a| a =~ Re0[:received] }
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # Unable to deliver message to: <kijitora@example.org>
+            # Delivery failed for the following reason:
+            # Server mx22.example.org[192.0.2.222] failed with: 550 <kijitora@example.org> No such user here
+            #
+            # This has been a permanent failure.  No further delivery attempts will be made.
+            v = dscontents[-1]
+
+            if cv = e.match(/\AUnable to deliver message to: [<]([^ ]+[@][^ ]+)[>]\z/)
+              # Unable to deliver message to: <kijitora@example.org>
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['recipient'] = cv[1]
+              recipients += 1
+
+            else
+              # Error message
+              if e =~ Re1[:error]
+                # Delivery failed for the following reason:
+                v['diagnosis'] = e
+
+              else
+                # Detect error message
+                next if e.empty?
+                next if v['diagnosis'].nil? || v['diagnosis'].empty?
+                next if e =~ /\A[-]+/
+
+                # Server mx22.example.org[192.0.2.222] failed with: 550 <kijitora@example.org> No such user here
+                v['diagnosis'] ||= ''
```

Comment:
__[rubocop]__
* Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.

<!-- Comment by [SideCI](https://sideci.com) -->